### PR TITLE
Ms3 security

### PR DESCRIPTION
**Pull request by VdotR**
Email Verification Ready. Should I create a helper tool for creating verified user for easier testing?

### CODE REVIEW

**Comment by cbaeucsd** [comment]
Also unless I missed it, give an opportunity for user to resend verification email.

**Comment by VdotR** [comment]
As of now verification email is sent upon logging in (check user.routes backend). How would user prompt the app to resend verification email?

**Comment by VdotR** [comment]
Only case I can think of is that the when am unverified user tries to login, we can reject and ask the user whether or not to send another verification email?

**Comment by cbaeucsd** [comment]
Remove lookup, signup and resend_verification use same code so simplify, remove token from get by id, remove leftover console.logs

File: backend/api/routes/user.routes.js [hunk 1]
```diff
@@ -218,5 +234,43 @@ router.patch("/change_password", checkSession, async (req, res) => {
...
+    const { email, subject, text } = req.body;
+
+    try {
+        await sendCustomEmail(email, subject, text);
+        res.send('Email sent successfully');
+    }
+    catch (error) {
+        res.status(400).send("Invalid request while sending email");
+    }
+});
```
[the remark this answers is not in the document]
Comment (cbaeucsd) [reply backend/api/routes/user.routes.js:313]: remove this

**Comment by VdotR** [comment]
Fixed

**Review by cbaeucsd — APPROVED** [review]

**Comment by cbaeucsd** [comment]
good

**Pull request merged**